homeassistant/components/sensor/__init__.py

```diff
@@ -390,6 +391,24 @@ def _get_initial_suggested_unit(self) -> str | UndefinedType:
         if suggested_unit_of_measurement is None:
             return UNDEFINED
 
+        # Make sure we can convert the units
+        if (
+            (unit_converter := UNIT_CONVERTERS.get(self.device_class)) is None
+            or self.unit_of_measurement not in unit_converter.VALID_UNITS
+            or suggested_unit_of_measurement not in unit_converter.VALID_UNITS
+        ):
+            if not self._invalid_suggested_unit_of_measurement_reported:
+                self._invalid_suggested_unit_of_measurement_reported = True
+                report_issue = self._suggest_report_issue()
+                # This should raise in Home Assistant Core 2025.2
+                _LOGGER.warning(
+                    "%s has an invalid suggested_unit_of_measurement. Please %s",
```

Comment:
Entity_id is not set/generated yet. Can we add something else?

Reply:
Let's print the type